### ISSUE TITLE
Metering panic under native engine after module deserialize

### ISSUE THREAD

**Issue by xofyarg** [wasmerio/wasmer#2234]
### Describe the bug

It seems that when putting the following 3 things together, wasm instance will panic:
1. use native engine, compile can be either LLVM or Cranelift
2. enable metering
3. serialize module then deserialize

It doesn't happen if use JIT engine or without module serialize/deserialize step 

```sh
$ rustc -V
rustc 1.47.0 (18bf6b4f0 2020-10-07)

$ uname -psmr
Linux 5.9.14 x86_64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
```

### Steps to reproduce

```rust
use wasmer::*;
use wasmer_middlewares::metering::get_remaining_points;

let wasm = wat2wasm(br#"(module)"#).unwrap();
let mut compiler_config = Cranelift::default();

let cost_function = |_: &Operator| -> u64 { 1 };
compiler_config.push_middleware(Arc::new(Metering::new(u64::MAX, cost_function)));

let engine = Native::new(compiler_config).engine();

let store = Store::new(&engine);
let module = Module::new(&store, &wasm).unwrap();

let bin = module.serialize().unwrap();
let module = unsafe { Module::deserialize(&store, &bin).unwrap() };

let imports = imports! {};

let instance = Instance::new(&module, &imports).unwrap();
let _ = get_remaining_points(&instance);
```

### Expected behavior
Run without error.

### Actual behavior
```
thread 'test::test_compile_and_load' panicked at 'Can't get `wasmer_metering_points_exhausted` from Instance: Missing("wasmer_metering_points_exhausted")', /build/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmer-middlewares-1.0.2/src/metering.rs:236:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by bobrik** [comment]
Seems that native loader [serializes](https://github.com/wasmerio/wasmer/blob/1.0.2/lib/engine-native/src/artifact.rs#L186) and [passes](https://github.com/wasmerio/wasmer/blob/1.0.2/lib/compiler-llvm/src/compiler.rs#L200-L201) metadata before [compiler applies middlewares](https://github.com/wasmerio/wasmer/blob/1.0.2/lib/compiler-llvm/src/compiler.rs#L203-L205). Naturally, after deserializing, there is no trace of middleware in the metadata. I'm not sure how to untangle this nicely.

**Comment by Hywan** [comment]
Thank you for the bug report. We are able to replicate the error.

**Comment by syrusakbary** [comment]
I believe this may be solved by #2916, we'll double check before closing this issue

**Comment by ptitSeb** [comment]
I adapted the minimum code to warmer 3.0 API
```rust
fn main() {
use std::sync::Arc;
use wasmer::*;
use wasmer::wasmparser::Operator;
use wasmer_middlewares::metering::get_remaining_points;
use wasmer_middlewares::Metering;

let wasm = wat2wasm(br#"(module)"#).unwrap();
let mut compiler_config = Cranelift::default();

let cost_function = |_: &Operator| -> u64 { 1 };
compiler_config.push_middleware(Arc::new(Metering::new(u64::MAX, cost_function)));

let mut store = Store::new(EngineBuilder::new(compiler_config));

let module = Module::new(&store, &wasm).unwrap();

let bin = module.serialize().unwrap();
let module = unsafe { Module::deserialize(&store, &bin).unwrap() };

let imports = imports! {};

let instance = Instance::new(&mut store, &module, &imports).unwrap();
let _ = get_remaining_points(&mut store, &instance);
}
```

And it works with no issue. Closing the issue.
